File: scanner.go
package main

import (
    "fmt"
    "net"
    "time"
)

func scanPort(address string, port int, timeout time.Duration) bool {
    conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", address, port), timeout)
    if err != nil {
        return false
    }
    conn.Close()
    return true
}

func main() {
    address := "example.com"
    timeout := time.Second * 3
    for port := 1; port <= 65535; port++ {
        if scanPort(address, port, timeout) {
            fmt.Printf("Port %d is open\n", port)
        }
    }
}

File: scanner.rs
use std::net::TcpStream;
use std::time::Duration;

fn scan_port(host: &str, port: u16, timeout: u64) -> bool {
    let addr = format!("{}:{}", host, port);
    match TcpStream::connect_timeout(&addr, Duration::from_secs(timeout)) {
        Ok(_) => true,
        Err(_) => false,
    }
}

fn main() {
    let host = "example.com";
    let timeout = 3; // in seconds
    for port in 1..65535 {
        if scan_port(host, port, timeout) {
            println!("Port {} is open", port);
        }
    }
}
